src/Renderer.ts

```diff
@@ -4,72 +4,131 @@ import {
   escape
 } from './helpers.ts';
 import type { MarkedOptions } from './MarkedOptions.ts';
+import type { Tokens } from './Tokens.ts';
+import type { _Parser } from './Parser.ts';
 
 /**
  * Renderer
  */
 export class _Renderer {
   options: MarkedOptions;
+  parser!: _Parser; // set by the parser
```

Comment:
Can we set this in the constructor instead?

Reply:
No because the Renderer is instantiated before the Parser any time a renderer extension is used. The tokenizer has the same problem with the lexer being set this way. 

This problem mostly exists since the Lexer and Parser  can be used in a static way `marked.lexer(options)`

Maybe we can require instance in the future

Reply:
> Maybe we can require instance in the future

That sounds good to me 👍